database/ffldb/db.go

```diff
@@ -2080,5 +2246,42 @@ func openDB(dbPath string, network wire.BitcoinNet, create bool) (database.DB, e
 
 	// Perform any reconciliation needed between the block and metadata as
 	// well as database initialization, if needed.
-	return reconcileDB(pdb, create)
+	if _, err := reconcileDB(pdb, create); err != nil {
+		return nil, err
+	}
+
+	if !create {
+		// Run all required data migrations
+		if err := migrateBlockIndex(pdb); err != nil {
+			return nil, err
+		}
+	}
+
+	// Lookup bucket IDs for internal buckets
+	err = pdb.View(func(dbTxUncast database.Tx) error {
```

Comment:
Since there are tests and this certainly works, this isn't a big deal, but I wonder if this would be better using the standard bucket machinery to get a bucket and then copy the ID so this doesn't really need to understand the underlying details?

Something like the following:

```Go
	// Lookup bucket IDs for internal buckets
	err = pdb.View(func(dbTx database.Tx) error {
		meta := dbTx.Metadata()
		b := meta.Bucket(blockIdx2BucketName)
		if b == nil {
			message := "Database is missing internal bucket: blockIdx2Bucket"
			return makeDbErr(database.ErrBucketNotFound, message, nil)
		}
		pdb.blockIdx2BucketID = b.(*bucket).id

		b = meta.Bucket(blockHashBucketName)
		if b == nil {
			message := "Database is missing internal bucket: blockHashBucket"
			return makeDbErr(database.ErrBucketNotFound, message, nil)
		}
		pdb.blockHashBucketID = b.(*bucket).id

		return nil
	})
	if err != nil {
		// Ignore error on close since no further recovery could be done
		// anyways.
		_ = pdb.Close()
		return nil, err
	}
```